ui_components/src/travel/Map.js

```diff
@@ -205,102 +283,257 @@ function MapGridLines({ tileWidth, tileHeight, stageOffsetX, stageOffsetY, stage
+                                                <rect x="45" y="0" rx="2" ry="2" width="10" height="5" style={{ fill: 'transparent', stroke: strokeColor, strokeWidth:'2'}} />
+                                            </g>
+                                          </svg>
+                                    </div>) : null
+                                    );
+                                })()
+                            }
+                        </div>
+                    </ReactTransitionGroup.CSSTransition>
+                ))}
+            </ReactTransitionGroup.TransitionGroup>
+        </div>
+    );
+}
+
 function MapNearbyPlayers({ scoutData, tileWidth, tileHeight, playerId, ranksToView }) {
     return (
         <div id="scout_locations" className='map_locations'>
             {scoutData
                 .filter(user => ranksToView[parseInt(user.rank_num)] === true)
                 .map((player, index) => (
-                (player.user_id != playerId) &&
+                    (player.user_id != playerId) &&
                     <div key={player.user_id}
-                    className={alignmentClass(player.alignment) + " " + visibilityClass(player.invulnerable)}
-                    style={{
-                        cursor: "pointer",
-                        transform: `translate3d(${(player.target_x - 1) * tileWidth}px, ${(player.target_y - 1) * tileHeight}px, 0)`,
-                        backfaceVisibility: "hidden",
-                        filter: "blur(0)",
+                        className='map_location'
+                        style={{
+                            cursor: "pointer",
+                            transform: `translate3d(${(player.target_x - 1) * tileWidth}px, ${(player.target_y - 1) * tileHeight}px, 0)`,
+                            backfaceVisibility: "hidden",
                         }}>
-                    <div className='map_locations_tooltip'>{player.user_name}</div>
-                </div>
-            ))}
+                        <div className='map_location_tooltip'>{player.user_name}</div>
+                        <div className={alignmentClassPlayer(player.alignment, player.village_id) + visibilityClass(player.invulnerable)}></div>
+                    </div>
+                ))}
+        </div>
+    );
+}
+
+function MapNearbyPatrols({ patrolData, tileWidth, tileHeight }) {
+    return (
+        <div id="patrol_locations" className='map_locations'>
+            {patrolData
+                .map((patrol, index) => (
+                    <div key={patrol.patrol_id + '_' + patrol.patrol_type}
+                        className='map_location'
+                        style={{
+                            cursor: "pointer",
+                            transform: `translate3d(${(patrol.target_x - 1) * tileWidth}px, ${(patrol.target_y - 1) * tileHeight}px, 0)`,
+                            backfaceVisibility: "hidden",
```

Comment:
Interestingly I didn't see any performance difference with this on/off. What do you see on your machine?

Reply:
I was having performance issues with hardware acceleration turned off originally, so this is probably only helpful for slow machines.